src/Build/Construction/Solution/ProjectInSolution.cs

```diff
@@ -83,7 +84,12 @@ public sealed class ProjectInSolution
         /// <summary>
         /// Characters that need to be cleansed from a project name.
         /// </summary>
-        private static readonly char[] s_charsToCleanse = { '%', '$', '@', ';', '.', '(', ')', '\'' };
+#if NET
+        private static readonly SearchValues<char> s_charsToCleanse = SearchValues.Create(
+#else
+        private static readonly char[] s_charsToCleanse = (
```

Comment:
oh man putting the dangling paren in this side of the `#if` is a great readability improvement over what I had been trying